spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/ContractDownloader.java

```diff
@@ -62,22 +61,24 @@ public ContractDownloader(StubDownloader stubDownloader,
 	}
 
 	/**
-	 * Downloads JAR containing all the contracts. Plugin configuration gets updated with
-	 * the inclusion pattern for the downloaded contracts. The JAR with the contracts
-	 * contains all the contracts for all the projects. We're interested only in its
-	 * subset.
-	 * @param config - Plugin configuration that will get updated with the inclusion
-	 * pattern
+	 * Downloads JAR containing all the contracts. The JAR with the contracts contains all
+	 * the contracts for all the projects. We're interested only in its subset.
 	 * @return location of the unpacked downloaded stubs
 	 */
-	public File unpackedDownloadedContracts(ContractVerifierConfigProperties config) {
```

Comment:
It's a public method, we should deprecate it instead of removing it.

Reply:
@marcingrzejszczak where is this rule applicable? I mean many classes are used only internally even if methods are public, so if I modify all usages within the project - do I still need to preserve an old method? Is there some module, which is considered public API, which must follow this rule?

Reply:
In general if a class is public people might reference it. That's why wherever I can I'm using the package scoped classes. Since this is a library, I have no knowledge of who is actually using this class and how.

> Is there some module, which is considered public API, which must follow this rule?

I'm assuming that all of the public classes follow that rule. At least I try to assume so ;)

Reply:
Oh, that's painful... I mean in general, people can use some class / method even if it's not public :) (reflection). So I thought it's usually good to draw line somewhere and say that these classes are part of the public API and are designed to be used externally and those are internal things / implementation details and can be modified. Otherwise refactoring is just impossible.

Reply:
> Oh, that's painful... I mean in general, people can use some class / method even if it's not public :) (reflection)

If someone uses reflection then there's not much I can do ;)

>  So I thought it's usually good to draw line somewhere and say that these classes are part of the public API and are designed to be used externally and those are internal things / implementation details and can be modified. Otherwise refactoring is just impossible.

That's a valid point. That's why we try not to have public API as much as possible.